src/test/kotlin/uk/gov/justice/hmpps/offendersearch/controllers/OffenderSearchPhraseAPIIntegrationTest.kt

```diff
@@ -187,8 +185,436 @@ class OffenderSearchPhraseAPIIntegrationTest {
+        "dd-M-yyyy",
+        "d/MM/yyyy",
+        "d-MM-yyyy",
+        "d/M/yyyy",
+        "d-M-yyyy"
+    ).map { LocalDate.parse("1988-01-06").format(DateTimeFormatter.ofPattern(it)) }
+
+
+    @ParameterizedTest
+    @MethodSource("dateOfBirthFormats")
+    internal fun `can match by date of birth in various formats`(dateOfBirth: String) {
+      hasSingleMatch(phrase = dateOfBirth, crn = "X99999")
+    }
+  }
+
+  @Nested
+  @Disabled("waiting for implementation")
+  @TestInstance(PER_CLASS)
+  inner class PNCNumberMatching {
+    @BeforeAll
+    internal fun setUp() {
+      loadOffenders(
+          OffenderReplacement(crn = "X00001"),
+          OffenderReplacement(
+              crn = "X99999",
+              pncNumber = "2018/0023456X"
+          ))
+    }
+
+    @Suppress("unused")
+    fun pncNumberFormats() = listOf(
+        "2018/0023456X",
+        "2018/0023456x",
+        "2018/023456X",
+        "2018/23456X",
+        "18/0023456X",
+        "18/0023456x",
+        "18/023456X",
+        "18/23456X")
+
+
+    @ParameterizedTest
+    @MethodSource("pncNumberFormats")
+    internal fun `can match by pnc number in various formats`(pncNumber: String) {
+      hasSingleMatch(phrase = pncNumber, crn = "X99999")
+    }
+  }
+
+  @Nested
+  @Disabled("waiting for implementation")
+  @TestInstance(PER_CLASS)
+  inner class FirstNamePrefixMatching {
+    @BeforeAll
+    internal fun setUp() {
+      @Suppress("SameParameterValue")
+      loadOffenders(
+          OffenderReplacement(crn = "X00001", firstName = "Xabier"),
+          OffenderReplacement(
+              crn = "X99999",
+              surname = "Gramsci",
+              firstName = "Antonio"
+          ))
+    }
+
+    @Suppress("unused")
+    fun firstNamePrefixes() = listOf(
+        "Antoni",
+        "Anton",
+        "Anto",
+        "Ant",
+        "An",
+        "A")
+
+
+    @ParameterizedTest
+    @MethodSource("firstNamePrefixes")
+    internal fun `can match first name using partial name but only when not matching all terms`(firstName: String) {
+      hasSingleMatch(phrase = firstName, crn = "X99999", matchAllTerms = false)
+      hasNoMatch(phrase = firstName, matchAllTerms = true)
+    }
+  }
+
+  @Nested
+  @Disabled("waiting for implementation")
+  @TestInstance(PER_CLASS)
+  inner class DeletedOffenders {
+    @Suppress("unused")
+    fun matchAllTerms() = listOf(false, true)
+
+    @BeforeAll
+    internal fun setUp() {
+      loadOffenders(
+          OffenderReplacement(
+              crn = "X99999",
+              deleted = false
+          ),
+          OffenderReplacement(
+              crn = "X88888",
+              deleted = true
+          ))
+    }
+
+    @ParameterizedTest
+    @MethodSource("matchAllTerms")
+    internal fun `will not find deleted offenders`(matchAllTerms: Boolean) {
+      hasNoMatch(phrase = "X99999", matchAllTerms = matchAllTerms)
+      hasSingleMatch(phrase = "X88888", crn = "X99999", matchAllTerms = matchAllTerms)
```

Comment:
This confused me a little - I thought you were searching on the crn, didn't realise it represents the expected result.  Might be worth renaming this and similar parameters?

Reply:
agreed